tests/test_designmatrix_noisepars.py

```diff
@@ -0,0 +1,33 @@
+import pytest
+
+from pint.models import get_model_and_toas
+from pinttestdata import datadir
+
+
+@pytest.fixture()
+def model_and_toas():
+    parfile = datadir / "J1713+0747_small.gls.par"
+    timfile = datadir / "J1713+0747_small.tim"
+    return get_model_and_toas(parfile, timfile)
+
+
+def test_designmatrix_no_free_noise_param(model_and_toas):
+    model, toas = model_and_toas
+
+    # This should work.
+    M = model.designmatrix(toas)
+
+
+def test_designmatrix_free_noise_params(model_and_toas):
```

Comment:
Please use `pytest.mark.parametrize` to automatically turn this into three tests, one for each value of `noiseparam`. This will ensure separate pass/failure for each of the three, and also simplify the test and avoid the need to try to reset the model at the end of each one.